### DISCUSSION TITLE
Question about drag and drop to show glb with the model viewer

### DISCUSSION THREAD

**Discussion by moonyuet** [google/model-viewer#3537] [Q&A]
Hi, I am doing some research on making ReactDnD 3d viewer with model viewer. But I face some challenge when I try to use DnD function as the model viewer doesn't show the 3D files.  I already did _file.preview = URL.createObjectURL(file)_  specifically for the model viewer src to locate the imported data. 
Does the model viewer require the glb files saved in some specific area/database to locate the source of 3d files?

[codesandbox](https://codesandbox.io/s/import-file-zz4367?file=/src/App.js)

**Comment by elalish** [comment]
Setting the `src` to an ObjectURL should work; I'd recommend simplifying your page to get this working in isolation and then add in the rest of your code slowly.

**Reply by moonyuet** [reply]
Hi elalish, i've done printing and test with the code in the sandbox by replacing DnD with simple `input`. The file is saved as the blob (it shows the links when I try to print and list `{file.preview} `with `p`) but I still didn't manage to get the src loaded with model-viewer. But if i test with `img` to load some image, my code works. I dont understand why it doesn't work with `model-viewer`. You can review the screenshot below.
![image](https://user-images.githubusercontent.com/64118225/174928017-555fae18-eeed-471c-9a63-8f77f617dc65.png)

This is the link of the forked sandbox with the simple input function from the previous one: https://codesandbox.io/s/import-file-forked-u1lc5n?file=/src/App.js

**Reply by elalish** [reply]
I'm seeing warnings in your console when I upload a GLB that are not coming from MV, so I think the problem is in the uploader.
